### ISSUE TITLE
`__match_args__` is ignored in .pyi files

### ISSUE THREAD

**Issue by lucascool12** [DetachHead/basedpyright#1705]
### Description

When creating a `.pyi` type stub file with a class that contains a `__match_args__` class attribute the `__match_args__` in the stub file gets ignored when using the class in a match statement.

For example if we create the following files:

```
foo/
├─ __main__.py
├─ __init__.py
├─ __init__.pyi
```

With the following contents:

<details>

<summary>__init__.py</summary>

```python
class Test:
    __match_args__: tuple[str, ...] = ("foo", "bar", "bop")
    def __init__(self):
        self.foo = "a"
        self.bar = "b"
        self.bop = "c"
```

</details>

<details>
<summary>__init__.pyi</summary>

```python
class Test:
    __match_args__: tuple[str, ...] = ("foo", "bar", "bop")
    foo: str
    bar: str
    bop: str
    def __init__(self) -> None: ...
```

</details>

And a `__main__.py` file with the following contents:

```python
from typing import reveal_type
from . import Test

b = Test()
match b:
    case Test(foo, bar, bop):
        reveal_type(foo)
        reveal_type(bar)
        reveal_type(bop)
```

This case statement is correct and runs using `python -m foo`. Both BasedPyright and Pyright report the variables `foo`, `bar` and `bop` as unknown and the case statement is reported as having too many positional patterns.

This issue was observed using BasedPyright version 1.37.1.

There doesn't seem to be a corresponding Pyright issue.

**Comment by DetachHead** [comment]
this is because the type annotation `tuple[str, ...]` is widening the type to a tuple of strings, so pyright no longer knows what the literal values are. it works if you type it like this:

```py
# foo/__init__.pyi
class Test:
    __match_args__: tuple[Literal["foo"], Literal["bar"], Literal["bop"]]
```

alternatively, it seems to also work if you remove the type annotation entirely, in which case it will infer the correct type from the value:

```py
# foo/__init__.pyi
class Test:
    __match_args__ = ("foo", "bar", "bop")
```

i wouldn't recommand that though because stubs aren't meant to have values in them, just type annotations. i suspect it's also less likely to work across other type checkers (if that's something you care about)

**Comment by lucascool12** [comment]
Thank you very much for the response. I'll type `__match_args__` like you recommend.

This is somewhat surprising behaviour, since no errors are giving except when you try to use it, and then it is not really clear that the problem is in the stub files. Perhaps a lint could be added that warns on stuff like this in stub files?
